### PR TITLE
Cleanup unused dependencies in cabal files

### PR DESCRIPTION
**Pull request by Jasagredo**
I happened to find that several dependencies were listed in cabal files but were not used at all. Removing them results in no harm.

Feel free to point to specific dependencies that are removed in this PR but you want to be kept as they will be used shortly.

### CODE REVIEW

File: ouroboros-network-framework/ouroboros-network-framework.cabal [hunk 1]
```diff
@@ -76,15 +76,11 @@ library
                      , contra-tracer
 
                      , io-classes      >=0.1   && < 0.3
-                     , monoidal-synchronisation
-                                       >=0.1   && < 0.2
```
Comment (coot) [inline ouroboros-network-framework/ouroboros-network-framework.cabal:80]: Could you leave it, it's used in `p2p-master` branch which sooner or later will be merged to `master`.

Comment (Jasagredo) [reply ouroboros-network-framework/ouroboros-network-framework.cabal]: Brought back, thanks!

**Review by nfrisby — APPROVED** [review]
LGTM.

As a follow-up, let's consider configuring `weeder` in our CI.

**Comment by Jasagredo** [comment]
> LGTM.
> 
> As a follow-up, let's consider configuring `weeder` in our CI.

I agree we should add weeder as a step on the CI. We just need to identify the "top" modules (or roots as weeder names them). But regarding unused dependencies, weeder can't help us there, it is only for finding dead code:
> Weeder is an application to perform whole-program dead-code analysis.

**Review by dcoutts — APPROVED** [review]
:tada:

**Pull request merged**